src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -6749,44 +7388,112 @@ void Assembler::xorl(Address dst, int32_t imm32) {
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  (void) evex_prefix_and_encode_ndd(src1->encoding(), dst->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  // opcode matches gcc
+  emit_arith(0x31, 0xC0, src1, src2);
+}
+
 void Assembler::xorl(Address dst, Register src) {
   InstructionMark im(this);
   prefix(dst, src);
   emit_int8(0x31);
   emit_operand(src, dst, 0);
 }
 
+void Assembler::exorl(Register dst, Address src1, Register src2, bool no_flags) {
+  InstructionMark im(this);
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  attributes.set_address_attributes(/* tuple_type */ EVEX_NOSCALE, /* input_size_in_bits */ EVEX_32bit);
+  evex_prefix_ndd(src1, dst->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  emit_int8(0x31);
+  emit_operand(src2, src1, 0);
+}
+
 void Assembler::xorb(Register dst, Address src) {
   InstructionMark im(this);
   prefix(src, dst);
   emit_int8(0x32);
   emit_operand(dst, src, 0);
 }
 
+void Assembler::exorb(Register dst, Register src1, Address src2, bool no_flags) {
+  InstructionMark im(this);
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  attributes.set_address_attributes(/* tuple_type */ EVEX_NOSCALE, /* input_size_in_bits */ EVEX_8bit);
+  evex_prefix_ndd(src2, dst->encoding(), src1->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  emit_int8(0x32);
+  emit_operand(src1, src2, 0);
+}
+
 void Assembler::xorb(Address dst, Register src) {
   InstructionMark im(this);
   prefix(dst, src, true);
   emit_int8(0x30);
   emit_operand(src, dst, 0);
 }
 
+void Assembler::exorb(Register dst, Address src1, Register src2, bool no_flags) {
+  InstructionMark im(this);
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  attributes.set_address_attributes(/* tuple_type */ EVEX_NOSCALE, /* input_size_in_bits */ EVEX_8bit);
+  evex_prefix_ndd(src1, dst->encoding(), src2->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  emit_int8(0x30);
+  emit_operand(src2, src1, 0);
+}
+
+void Assembler::xorw(Register dst, Register src) {
+  (void)prefix_and_encode(dst->encoding(), src->encoding());
+  emit_arith(0x33, 0xC0, dst, src);
+}
+
+void Assembler::exorw(Register dst, Register src1, Register src2, bool no_flags) {
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  (void) evex_prefix_and_encode_ndd(src1->encoding(), dst->encoding(), src2->encoding(), VEX_SIMD_66, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  emit_arith(0x33, 0xC0, src1, src2);
+}
+
```

Comment:
We just removed `xorw(reg, reg)` because it is not used: [#20901](https://git.openjdk.org/jdk/pull/20901)
Please, don't add it back.

Reply:
Sorry.  Done.